test/autogen/test_math_user_proxy_agent.py

```diff
@@ -0,0 +1,87 @@
+from flaml import oai
+from flaml.autogen.agent.math_user_proxy_agent import MathUserProxyAgent
+import pytest
+import sys
+
+KEY_LOC = "test/autogen"
+
+
+@pytest.mark.skipif(
+    sys.platform in ["darwin", "win32"],
+    reason="do not run on MacOS or windows",
+)
+def test_math_user_proxy_agent():
+    try:
+        import openai
+    except ImportError:
+        return
+
+    from flaml.autogen.agent.assistant_agent import AssistantAgent
+
+    conversations = {}
+    oai.ChatCompletion.start_logging(conversations)
+
+    config_list = oai.config_list_openai_aoai(key_file_path=KEY_LOC)
+    assistant = AssistantAgent(
+        "assistant",
+        system_message="You are a helpful assistant.",
+        request_timeout=600,
+        seed=42,
+        config_list=config_list,
+    )
+
+    mathproxyagent = MathUserProxyAgent(name="MathChatAgent", human_input_mode="NEVER")
+    assistant.reset()
+
+    math_problem = "$x^3=125$. What is x?"
+    assistant.receive(
+        message=mathproxyagent.generate_prompt(math_problem),
```

Comment:
Not all the code paths in `generate_prompt` are covered.
Also, consider adding test of the new notebook in test/autogen/oai/test_notebook (maybe it's better to move it to test/autogen/)

Reply:
I have added the 'two_tools' prompt, and added this notebook in test.

Reply:
Where is the "two_tools" prompt added?